<a href="https://colab.research.google.com/github/kim-kidong/hello-world/blob/master/3_2_Pandas_Practice_ipynb%EC%9D%98_%EC%82%AC%EB%B3%B8%EC%9D%98_%EC%82%AC%EB%B3%B8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd

#Q1. DataFrame 생성
---
`d`를 사용하여 Pandas DataFrame을 생성하고 `df`라는 이름으로 저장하세요.

In [ ]:
# 문제의 입력값입니다. 수정하지 마세요.
d = {
    "이름" : [
        "김변전",
        "박사무",
        "장건축"
    ],
    "직급" : [
        "4(가)",
        "3",
        "4(나)"
    ],
    "근속년수" : [
        20,
        18,
        3
    ]
}
print(d)

In [ ]:
# 아래에 정답코드를 작성해주세요.
df = pd.DataFrame(d)
print(df)

#Q2. Series 생성
---
다음 조건에 맞는 Pandas Series를 생성하고 `ps`라는 이름으로 저장하세요.  
Name : favorite_fruit  
Objects : "apple", "grape", "kiwi"

In [ ]:
# 아래에 정답코드를 작성해주세요.

ps = pd.Series(["apple", "grape", "kiwi"], name = "favorite_fruit")
print(ps)

0    apple
1    grape
2     kiwi
Name: favorite_fruit, dtype: object


#Q3. DataFrame 분석
---
Q1. 에서 생성한 `df`의 "근속년수" 컬럼 중 가장 큰 값을 출력하세요.

In [ ]:
# 아래에 정답코드를 작성해주세요.

df["근속년수"].mean()

13.666666666666666

#Q4. 외부 테이블 데이터 읽기
---
주어진 URL은 csv파일 다운로드 경로입니다. 해당 파일을 DataFrame으로 읽어와 `titanic` 이라는 이름으로 저장하세요.  
URL : https://docs.google.com/uc?export=download&id=17ACmFtAlNNi_wZgGgw2Vpms3zfZrVJw6  
*Hint. pd.read_csv()*

In [ ]:
# 아래에 정답코드를 작성해주세요.

titanic = pd.read_csv("https://docs.google.com/uc?export=download&id=17ACmFtAlNNi_wZgGgw2Vpms3zfZrVJw6")

#Q5. DataFrame 출력
---
`titanic`의 맨 위 10행만 출력하세요.

In [ ]:
# 아래에 정답코드를 작성해주세
titanic.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

#Q6. 열 필터링
---
`titanic`에서 "Name", "Sex" 열만 선택하여 출력하세요.

In [ ]:
# 아래에 정답코드를 작성해주세요.

titanic[["Name", "Sex"]]

Name     Sex
0                              Braund, Mr. Owen Harris    male
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female
2                               Heikkinen, Miss. Laina  female
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female
4                             Allen, Mr. William Henry    male
..                                                 ...     ...
886                              Montvila, Rev. Juozas    male
887                       Graham, Miss. Margaret Edith  female
888           Johnston, Miss. Catherine Helen "Carrie"  female
889                              Behr, Mr. Karl Howell    male
890                                Dooley, Mr. Patrick    male

[891 rows x 2 columns]

#Q7. 행 필터링(1)
---
`titanic`에서 남성 승객의 정보만 선택하여 `titanic_male`이라는 이름으로 저장하고 출력하세요.  
*Hint. DataFrame["column"], "Sex"열의 값이 "male"이면 남성*

In [ ]:
# 아래에 정답코드를 작성해주세요.
titanic_male = titanic["Sex"] == "male"
titanic[titanic_male]


PassengerId  Survived  Pclass                            Name   Sex  \
0              1         0       3         Braund, Mr. Owen Harris  male   
4              5         0       3        Allen, Mr. William Henry  male   
5              6         0       3                Moran, Mr. James  male   
6              7         0       1         McCarthy, Mr. Timothy J  male   
7              8         0       3  Palsson, Master. Gosta Leonard  male   
..           ...       ...     ...                             ...   ...   
883          884         0       2   Banfield, Mr. Frederick James  male   
884          885         0       3          Sutehall, Mr. Henry Jr  male   
886          887         0       2           Montvila, Rev. Juozas  male   
889          890         1       1           Behr, Mr. Karl Howell  male   
890          891         0       3             Dooley, Mr. Patrick  male   

      Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0    22.0      1      0         A/5 21171   7.2500   NaN        S  
4    35.0      0      0            373450   8.0500   NaN        S  
5     NaN      0      0            330877   8.4583   NaN        Q  
6    54.0      0      0             17463  51.8625   E46        S  
7     2.0      3      1            349909  21.0750   NaN        S  
..    ...    ...    ...               ...      ...   ...      ...  
883  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
886  27.0      0      0            211536  13.0000   NaN        S  
889  26.0      0      0            111369  30.0000  C148        C  
890  32.0      0      0            370376   7.7500   NaN        Q  

[577 rows x 12 columns]

#Q8. 행 필터링(2)
---
`titanic`에서 생존한 여성 승객의 정보만 선택하여 `titanic_s_f`이라는 이름으로 저장하고 출력하세요.  
*Hint. "Survived"열의 값이 1이면 생존*

In [ ]:
# 아래에 정답코드를 작성해주세요.
titanic_s_f = (titanic["Sex"] == "female") & (titanic["Survived"] == 1)
titanic[titanic_s_f]

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
874          875         1       2   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
874              Abelson, Mrs. Samuel (Hannah Wizosky)  female  28.0      1   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
874      0         P/PP 3381  24.0000   NaN        C  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  

[233 rows x 12 columns]

#Q9. 행,열 필터링(1)
---
`titanic`에서 100-120번 행, 0-5번 열만 선택하여 출력하세요.(index 기준)  
*Hint. .iloc[:,:]*

In [ ]:
# 아래에 정답코드를 작성해주세요.
titanic.iloc[100:121, 0:6]


PassengerId  Survived  Pclass                               Name     Sex  \
100          101         0       3            Petranec, Miss. Matilda  female   
101          102         0       3   Petroff, Mr. Pastcho ("Pentcho")    male   
102          103         0       1          White, Mr. Richard Frasar    male   
103          104         0       3         Johansson, Mr. Gustaf Joel    male   
104          105         0       3     Gustafsson, Mr. Anders Vilhelm    male   
105          106         0       3              Mionoff, Mr. Stoytcho    male   
106          107         1       3   Salkjelsvik, Miss. Anna Kristine  female   
107          108         1       3             Moss, Mr. Albert Johan    male   
108          109         0       3                    Rekic, Mr. Tido    male   
109          110         1       3                Moran, Miss. Bertha  female   
110          111         0       1     Porter, Mr. Walter Chamberlain    male   
111          112         0       3               Zabour, Miss. Hileni  female   
112          113         0       3             Barton, Mr. David John    male   
113          114         0       3            Jussila, Miss. Katriina  female   
114          115         0       3              Attalah, Miss. Malake  female   
115          116         0       3              Pekoniemi, Mr. Edvard    male   
116          117         0       3               Connors, Mr. Patrick    male   
117          118         0       2    Turpin, Mr. William John Robert    male   
118          119         0       1           Baxter, Mr. Quigg Edmond    male   
119          120         0       3  Andersson, Miss. Ellis Anna Maria  female   
120          121         0       2        Hickman, Mr. Stanley George    male   

      Age  
100  28.0  
101   NaN  
102  21.0  
103  33.0  
104  37.0  
105  28.0  
106  21.0  
107   NaN  
108  38.0  
109   NaN  
110  47.0  
111  14.5  
112  22.0  
113  20.0  
114  17.0  
115  21.0  
116  70.5  
117  29.0  
118  24.0  
119   2.0  
120  21.0

#Q10. 행,열 필터링(2)
---
`titanic`에서 나이가 20 이상인 승객 정보 중 "Name", "Sex" 정보만 선택하여 출력하세요.  
*Hint. .loc[]

In [ ]:
# 아래에 정답코드를 작성해주세요.

is_more_20 = titanic["Age"] >= 20
titanic.loc[is_more_20, [ "Name", "Sex"]]

Name     Sex
0                              Braund, Mr. Owen Harris    male
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female
2                               Heikkinen, Miss. Laina  female
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female
4                             Allen, Mr. William Henry    male
..                                                 ...     ...
884                             Sutehall, Mr. Henry Jr    male
885               Rice, Mrs. William (Margaret Norton)  female
886                              Montvila, Rev. Juozas    male
889                              Behr, Mr. Karl Howell    male
890                                Dooley, Mr. Patrick    male

[550 rows x 2 columns]

#Q11. 열 수정(1)
---
`titanic` 승객의 나이 정보가 오류로 인해 1살씩 어리게 기록되었습니다. 이를 올바로 복구하여 저장하세요.

In [ ]:
# 아래에 정답코드를 작성해주세요.
titanic["Age"] = titanic["Age"] + 1
titanic.head(5)


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  23.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  39.0      1   
2                             Heikkinen, Miss. Laina  female  27.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  36.0      1   
4                           Allen, Mr. William Henry    male  36.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

#Q12. 열 수정(2)
---
`titanic` 침몰 후 109년이 지났습니다. `titanic` 승객의 생존 시 현재 나이를 "AgeNow"라는 열을 만들어 저장하세요.

In [ ]:
# 아래에 정답코드를 작성해주세요.
titanic["AgeNow"] = titanic["Age"] + 110
titanic.head(5)


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  23.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  39.0      1   
2                             Heikkinen, Miss. Laina  female  27.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  36.0      1   
4                           Allen, Mr. William Henry    male  36.0      0   

   Parch            Ticket     Fare Cabin Embarked  AgeNow  
0      0         A/5 21171   7.2500   NaN        S   133.0  
1      0          PC 17599  71.2833   C85        C   149.0  
2      0  STON/O2. 3101282   7.9250   NaN        S   137.0  
3      0            113803  53.1000  C123        S   146.0  
4      0            373450   8.0500   NaN        S   146.0

#Q13. 통계, 집계(1)
---
`titanic` 승객의 좌석등급("Pclass")의 평균을 출력하세요.  
*Hint. .mean()*

In [ ]:
# 아래에 정답코드를 작성해주세요.

titanic["Pclass"].mean()

2.308641975308642

#Q14. 통계, 집계(2)
`titanic` 승객의 성별 생존률을 출력하세요.  
*Hint. groupby, .mean()*

In [ ]:
# 아래에 정답코드를 작성해주세요.

titanic.groupby("Sex")["Survived"].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

#Q15. 행 정렬
---
`titanic`을 승객의 연령 내림차순으로 정렬하여 `titanic_desc` 라는 이름으로 저장하세요.  
*Hint. .sort_values(by=, ascending=)*

In [ ]:
# 아래에 정답코드를 작성
titanic.sort_values(by = "Age", ascending=False)



PassengerId  Survived  Pclass                                      Name  \
630          631         1       1      Barkworth, Mr. Algernon Henry Wilson   
851          852         0       3                       Svensson, Mr. Johan   
493          494         0       1                   Artagaveytia, Mr. Ramon   
96            97         0       1                 Goldschmidt, Mr. George B   
116          117         0       3                      Connors, Mr. Patrick   
..           ...       ...     ...                                       ...   
859          860         0       3                          Razi, Mr. Raihed   
863          864         0       3         Sage, Miss. Dorothy Edith "Dolly"   
868          869         0       3               van Melkebeke, Mr. Philemon   
878          879         0       3                        Laleff, Mr. Kristo   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   

        Sex   Age  SibSp  Parch      Ticket     Fare Cabin Embarked  AgeNow  
630    male  81.0      0      0       27042  30.0000   A23        S   191.0  
851    male  75.0      0      0      347060   7.7750   NaN        S   185.0  
493    male  72.0      0      0    PC 17609  49.5042   NaN        C   182.0  
96     male  72.0      0      0    PC 17754  34.6542    A5        C   182.0  
116    male  71.5      0      0      370369   7.7500   NaN        Q   181.5  
..      ...   ...    ...    ...         ...      ...   ...      ...     ...  
859    male   NaN      0      0        2629   7.2292   NaN        C     NaN  
863  female   NaN      8      2    CA. 2343  69.5500   NaN        S     NaN  
868    male   NaN      0      0      345777   9.5000   NaN        S     NaN  
878    male   NaN      0      0      349217   7.8958   NaN        S     NaN  
888  female   NaN      1      2  W./C. 6607  23.4500   NaN        S     NaN  

[891 rows x 13 columns]

#Q16. DataFrame 병합
---
`df0110`은 2001-2010년, `df1120`은 2011-2020년 용도별 전력 판매량 입니다.  
두 DataFrame을 병합하여 `df0120`이라는 이름으로 저장하고 출력하세요.  
*Hint. pd.concat()*

In [ ]:
# 문제의 입력값입니다. 수정하지 마세요.
url = "https://docs.google.com/uc?export=download&id="
df0110 = pd.read_csv(
    url + "1_lhuWUoEk05YMtw9LuuLjOOH_ahVBsfh", encoding="cp949"
)
df1120 = pd.read_csv(
    url + "1j6X33UDx1of6Eyp8ZaH86A-aACtmV_9H", encoding="cp949"
)


In [ ]:
# 아래에 정답코드를 작성해주세요.
df0120 = pd.concat([df0110 , df1120]).sort_values("연도")

df0120.reset_index(drop=True)

연도           주택용           일반용          교육용           산업용           농사용  \
0   2001  4.026973e+07  5.262244e+07  2639687.250  1.421602e+08  6.142090e+06   
1   2002  4.345214e+07  5.742801e+07  2953566.377  1.511962e+08  6.328169e+06   
2   2003  4.650535e+07  6.162642e+07  3351320.214  1.578445e+08  6.147364e+06   
3   2004  4.951330e+07  6.747597e+07  3773549.221  1.662230e+08  6.766324e+06   
4   2005  5.204446e+07  7.371552e+07  4308747.515  1.749454e+08  7.317815e+06   
5   2006  5.391227e+07  7.780915e+07  4790496.283  1.830671e+08  7.635660e+06   
6   2007  5.568085e+07  8.220759e+07  5304103.629  1.949364e+08  8.215174e+06   
7   2008  5.787751e+07  8.682700e+07  5783323.517  2.034746e+08  8.869459e+06   
8   2009  5.942725e+07  8.961932e+07  6465448.526  2.072159e+08  9.671357e+06   
9   2010  6.319960e+07  9.741003e+07  7452576.466  2.326722e+08  1.065429e+07   
10  2011  6.352366e+07  9.950406e+07  7568016.011  2.514906e+08  1.123154e+07   
11  2012  6.548373e+07  1.015928e+08  7860449.360  2.581019e+08  1.277604e+07   
12  2013  6.581457e+07  1.021963e+08  7947317.399  2.653725e+08  1.386584e+07   
13  2014  6.445726e+07  1.007609e+08  7438373.754  2.725516e+08  1.450473e+07   
14  2015  6.561861e+07  1.036793e+08  7690600.115  2.735480e+08  1.570247e+07   
15  2016  6.805679e+07  1.086169e+08  8079392.000  2.788279e+08  1.657984e+07   
16  2017  6.854376e+07  1.112978e+08  8316290.000  2.859696e+08  1.725085e+07   
17  2018  7.289471e+07  1.169339e+08  8677808.000  2.929987e+08  1.850383e+07   
18  2019  7.263887e+07  1.162265e+08  8560660.333  2.892402e+08  1.888237e+07   
19  2020  7.630340e+07  1.136385e+08  7515352.000  2.786602e+08  1.902883e+07   

            가로등            심야            합계  
0   1877624.402  1.201957e+07  2.577314e+08  
1   2075944.013  1.501735e+07  2.784514e+08  
2   2197210.383  1.592705e+07  2.935992e+08  
3   2367076.988  1.597638e+07  3.120956e+08  
4   2570005.506  1.751091e+07  3.324128e+08  
5   2687040.040  1.881762e+07  3.487194e+08  
6   2793943.083  1.946737e+07  3.686054e+08  
7   2847240.868  1.939099e+07  3.850701e+08  
8   2954230.980  1.912116e+07  3.944746e+08  
9   3081490.265  1.969006e+07  4.341602e+08  
10  3145497.971  1.860684e+07  4.550703e+08  
11  3158091.480  1.761994e+07  4.665929e+08  
12  3156225.641  1.649576e+07  4.748486e+08  
13  3221008.114  1.465787e+07  4.775917e+08  
14  3341239.614  1.407456e+07  4.836548e+08  
15  3462314.000  1.341583e+07  4.970389e+08  
16  3556775.000  1.281137e+07  5.077464e+08  
17  3582649.000  1.255759e+07  5.261492e+08  
18  3571475.434  1.137866e+07  5.204987e+08  
19  3506999.000  1.061634e+07  5.092697e+08

#Q17. DataFrame join
---
*air_quality_pm25* 는 미세먼지, *air_quality_no2*는 이산화질소 측정 값입니다.  
*air_quality_pm25*의 "date.utc", "location" 컬럼 기준으로 두 DataFrame을 join 하고 `air_quality`라는 이름으로 저장하세요.  
*Hint. pd.merge() how=[left || right], on="컬럼"*

In [ ]:
# 문제의 입력값입니다. 수정하지 마세요.
url = "https://docs.google.com/uc?export=download&id="
air_quality_pm25 = pd.read_csv(
    url + "1O2o5h277w6MoBYjbgOHCPtfbhABZ8XRB",
    parse_dates=True,
    usecols=["date.utc", "location", "parameter", "value"]
)

air_quality_no2 = pd.read_csv(
    url + "1nc4toKu_5ywrcD71Log_EYlZP6-6tby7",
    parse_dates=True,
    usecols=["date.utc", "location", "parameter", "value"]
)

In [ ]:
# 아래에 정답코드를 작성해주세요.

pd.merge(air_quality_pm25, air_quality_no2, how="right", on = ["date.utc", "location" ])

date.utc            location parameter_x  value_x  \
0     2019-06-21 00:00:00+00:00             FR04014         NaN      NaN   
1     2019-06-20 23:00:00+00:00             FR04014         NaN      NaN   
2     2019-06-20 22:00:00+00:00             FR04014         NaN      NaN   
3     2019-06-20 21:00:00+00:00             FR04014         NaN      NaN   
4     2019-06-20 20:00:00+00:00             FR04014         NaN      NaN   
...                         ...                 ...         ...      ...   
2063  2019-05-07 06:00:00+00:00  London Westminster        pm25      9.0   
2064  2019-05-07 04:00:00+00:00  London Westminster        pm25      8.0   
2065  2019-05-07 03:00:00+00:00  London Westminster        pm25      8.0   
2066  2019-05-07 02:00:00+00:00  London Westminster        pm25      8.0   
2067  2019-05-07 01:00:00+00:00  London Westminster        pm25      8.0   

     parameter_y  value_y  
0            no2     20.0  
1            no2     21.8  
2            no2     26.5  
3            no2     24.9  
4            no2     21.4  
...          ...      ...  
2063         no2     26.0  
2064         no2     16.0  
2065         no2     19.0  
2066         no2     19.0  
2067         no2     23.0  

[2068 rows x 6 columns]

#Q18. 시계열 데이터(Time Series)
---
`air_quality`의 "date.utc" 컬럼을  Pandas의 datetime dtype으로 변경하세요.

In [ ]:
1# 아래에 정답코드를 작성해주세요.

pd.to_datetime(air_quality["date.utc"])

KeyError: ignored

#Q19. Resample
---
`air_quality`를 일별 미세먼지, no2수치 합으로 resample 하세요.  
*Hint. resample하기 위해서는 datetime이 index여야 합니다. `sum()`*

In [ ]:
# 아래에 정답코드를 작성해주세요.

air_quality.set_index("date.utc", inplace = True)


KeyError: ignored

In [ ]:
air_quality.resample("D").sum().head()